### PR TITLE
Arc 2596 jenkins setup screen

### PR DESCRIPTION
**Pull request by rachellerathbone**
Same as https://github.com/atlassian/jenkins-for-jira/pull/230 but let's merge this one to main shall we? :sigh:

### CODE REVIEW

File: app/jenkins-for-jira-ui/src/api/fetchGlobalPageUrl.ts [hunk 1]
```diff
@@ -1,5 +1,20 @@
 import { invoke } from '@forge/bridge';
 
+type FetchSiteNameProps = {
+	withProtocol: boolean
+};
+
```
Comment (joshkay10) [inline app/jenkins-for-jira-ui/src/api/fetchGlobalPageUrl.ts:6]: /nit - we refer to is as siteName and siteUrl, can we rename the function to fetchSiteUrl

File: app/jenkins-for-jira-ui/src/api/fetchGlobalPageUrl.ts [hunk 2]
```diff
@@ -1,5 +1,20 @@
...
+const fetchSiteName = async ({ withProtocol }: FetchSiteNameProps): Promise<string> => {
```
Comment (joshkay10) [inline app/jenkins-for-jira-ui/src/api/fetchGlobalPageUrl.ts:7]: /nit - as far as i can tell this is only called once in JenkinsSetup.ts passing withProtocol:false
 ....moving from nit to opinion piece now lol. 
 I would remove the flag arg entirely, it muddies the code and isn't adding any extra benefits, if the time comes that we need a protocolless we should make fetchSiteNameWithoutProtocol which could probably just leverage existingg....
 
```
 fetchSiteNameWithoutProtocol() {
  siteUrl = await fetchSiteUrl()
  protocolRegx = /^https?:\/\//
  return siteUrl.replace(protocolregex, '');
  }
```

File: app/jenkins-for-jira-ui/src/components/JenkinsSetup/JenkinsSetup.tsx [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+						handleCopyToClipboard(copyRef);
+						setIsCopied(true);
+					}
+				}}
+				testId={testId}
+			>
+				Copy
+			</Button>
+
+			{isCopied ? <CopiedToClipboard /> : null}
```
Comment (joshkay10) [inline app/jenkins-for-jira-ui/src/components/JenkinsSetup/JenkinsSetup.tsx:86]: /nit can we simplify this? seems hard work to have an else null
`{isCopied && <CopiedToClipboard />}`

File: app/jenkins-for-jira-ui/src/components/JenkinsSetup/JenkinsSetup.tsx [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+};
+
+const IAmTheJenkinsAdmin = ({
+	handleCopyToClipboard,
+	webhookUrlRef,
+	secretRef
+}: CopyProps & IAmTheJenkinsAdminProps): JSX.Element => {
+	const handleFollowLink = (e: React.MouseEvent): void => {
+		e.preventDefault();
+		router.open('https://app.contentful.com/spaces/zsv3d0ugroxu/entries/4ccCes4jpnMlSVtI4Eqre9?focusedField=body');
```
Comment (joshkay10) [inline app/jenkins-for-jira-ui/src/components/JenkinsSetup/JenkinsSetup.tsx:153]: /nit - not now, but might be worth creating consts up top for these contentful links, just so we know what they are and easier to find/replace

File: app/jenkins-for-jira-ui/src/api/fetchGlobalPageUrl.ts (hunk 1 above)
Comment (rachellerathbone) [reply app/jenkins-for-jira-ui/src/api/fetchGlobalPageUrl.ts:6]: I'll create a ticket to standardise this. It started out with Dom calling it a site url then later switching to siteName, hence the naming here.

File: app/jenkins-for-jira-ui/src/components/JenkinsSetup/JenkinsSetup.tsx (hunk 4 above)
Comment (rachellerathbone) [reply app/jenkins-for-jira-ui/src/components/JenkinsSetup/JenkinsSetup.tsx:153]: lol I did actually do that in the other file where it's being used. Let me clean that up.

**Review by joshkay10 — APPROVED** [review]

**Pull request merged**